### ISSUE TITLE
Cannot create FutureParam

### ISSUE THREAD

**Issue by DarwinAwardWinner** [HenrikBengtsson/BiocParallel.FutureParam#2]
It seems like there might be a backward-incompatible change in BiocParallel that is breaking this package. I think `catch.errors` is no longer an argument to the param constructor, so this error happens whenever I try to create a FutureParam:

```
> BiocParallel.FutureParam::FutureParam()
Error in envRefSetField(.Object, field, classDef, selfEnv, elements[[field]]) : 
  ‘catch.errors’ is not a field in class “FutureParam”
> traceback()
10: stop(gettextf("%s is not a field in class %s", sQuote(field), 
        dQuote(thisClass@className)), domain = NA)
9: envRefSetField(.Object, field, classDef, selfEnv, elements[[field]])
8: initRefFields(.self, .refClassDef, as.environment(.self), list(...))
7: callSuper(...)
6: .Object$initialize(...)
5: initialize(value, ...)
4: initialize(value, ...)
3: new("FutureParam", ...)
2: .FutureParam(workers = 1L, catch.errors = catch.errors, stop.on.error = stop.on.error, 
       log = log, threshold = threshold, logdir = logdir)
1: BiocParallel.FutureParam::FutureParam()
```

Session info:

```
> sessionInfo()                                                                                          
R version 3.6.1 (2019-07-05)                                                                             
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.2 LTS
 
Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so
 
locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] rctutils_0.1.0

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.5               remotes_2.1.0
 [3] listenv_0.7.0                  purrr_0.3.2
 [5] tcltk_3.6.1                    rcmdcheck_1.3.3
 [7] colorspace_1.4-1               testthat_2.1.1
 [9] usethis_1.5.1                  stats4_3.6.1
[11] rlang_0.4.0                    pkgbuild_1.0.3
[13] pillar_1.4.2                   glue_1.3.1
[15] withr_2.1.2                    BiocParallel_1.18.0
[17] BiocGenerics_0.30.0            xopen_1.0.0
[19] sessioninfo_1.1.1              foreach_1.5.1
[21] stringr_1.4.0                  munsell_0.5.0
[23] commonmark_1.7                 gtable_0.3.0
[25] future_1.14.0                  devtools_2.1.0
[27] codetools_0.2-16               memoise_1.1.0
[29] callr_3.3.0                    IRanges_2.18.1
[31] ps_1.3.0                       parallel_3.6.1
[33] curl_3.3                       Rcpp_1.0.1
[35] backports_1.1.4                scales_1.0.0
[37] S4Vectors_0.22.0               desc_1.2.0
[39] pkgload_1.0.2                  doFuture_0.8.0
[41] fs_1.3.1                       ggplot2_3.2.0
[43] digest_0.6.20                  stringi_1.4.3
[45] processx_3.4.0                 dplyr_0.8.3
[47] rprojroot_1.3-2                grid_3.6.1
[49] cli_1.1.0                      tools_3.6.1
[51] magrittr_1.5                   lazyeval_0.2.2
[53] tibble_2.1.3                   crayon_1.3.4
[55] pkgconfig_2.0.2                BiocParallel.FutureParam_0.1.1
[57] xml2_1.2.0                     prettyunits_1.0.2
[59] iterators_1.0.10               assertthat_0.2.1
[61] roxygen2_6.1.1                 rstudioapi_0.10
[63] globals_0.12.4                 R6_2.4.0
[65] compiler_3.6.1
```

**Comment by DarwinAwardWinner** [comment]
I removed the offending argument and ran into a different issue: it looks like all the param types in BiocParallel itself now dynamically build a list of arguments, starting with `BiocParallel::.BiocParallelParam_prototype`, using `BiocParallel::.prototype_update` to add/override args, and then using `do.call` to call the constructor with that argument list. Failing to do so means that you end up with a bunch of missing arguments, which in turn means that `validObject` throws an error. I'm working on a pull request to make `FutureParam` follow this pattern.

**Comment by HenrikBengtsson** [comment]
Fixed in v0.2.0.
